cmd/plugins/juju-metadata/imagemetadata.go

```diff
@@ -39,20 +40,23 @@ func prepare(context *cmd.Context, controllerName string, store jujuclient.Clien
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	if _, ok := provider.(caas.ContainerEnvironProvider); ok {
```

Comment:
This would mean that none of the 'juju metadata' commands that need access to a provider will work for CAAS. Is this correct?

Reply:
yeah, for now.
We can test and enable later if we wanna support this for a k8s controller.